components/camel-optaplanner/src/main/java/org/apache/camel/component/optaplanner/OptaPlannerProducer.java

```diff
@@ -113,6 +143,54 @@ public void run() {
         }
     }
 
+    /**
+     * Using SolverManager
+     *
+     * @param  exchange
+     * @param  body
+     * @throws Exception
+     */
+    private void processWithSolverManager(Exchange exchange, Object body, AsyncCallback callback)
+            throws Exception {
+        final SolverManager solverManager = getSolverManager(exchange);
+
+        if (body.getClass().isAnnotationPresent(PlanningSolution.class)) {
+            LOGGER.debug("Asynchronously solving problem: [{}] with id [{}]", body);
+            Long problemId = endpoint.getConfiguration().getProblemId();
+            if (isAsync(exchange)) {
+                executor.submit(() -> {
+                    try {
+                        // create a consumer for best solution
+                        OptaplannerSolution consumer = endpoint.getOrCreateSolution(problemId);
+                        consumer.bestSolution(body);
+                        // start solving :: Solver Job is a thread
+                        SolverJob solverJob = solverManager.solveAndListen(problemId, t -> body, consumer::bestSolution);
+                        // wait for result
+                        populateResultWithSolverManager(exchange, solverJob);
+                    } catch (Throwable e) {
+                        LOGGER.error("Asynchronously solving failed with SolverManager, for problemId ({})", problemId, e);
+                    } finally {
+                        callback.done(false);
```

Comment:
Yes this is correct, after we have the response we can tell Camel to continue by calling false on the callback